connection.js

```diff
@@ -920,56 +819,74 @@ function pluralFamilyStringToSingularString(plural) { /* {{{ */
+function familyArrayFromFamilyNameString(familyNameString) { /* {{{ */
+    return ["", "ranked", "unranked"].map(e => familyNameString + e);
+} /* {{{ */
+
+function familyObjectMIBLIsminmax(familyNameString) { /* {{{ */
     let mm = "";
     let ir = "";
     let ba = "";
     let lh = "";
-    const isMin = generalArgString.includes("min");
-    const isMax = generalArgString.includes("max");
+    const isMin = familyNameString.includes("min");
+    const isMax = familyNameString.includes("max");
     if (isMin) {
         mm = "Minimum";
         ir = "increase";
         ba = "below";
         lh = "low";
-    } else if (isMax) {
+    } else {
         mm = "Maximum";
         ir = "reduce";
         ba = "above";
         lh = "high";
     }
-    let fullArray = ["", "ranked", "unranked"].map(e => generalArgString + e);
-    fullArray.push([mm, ir, ba, lh], [isMin, isMax]);
-    return fullArray;
+    const familyArray = familyArrayFromFamilyNameString(familyNameString);
+    return {argNameStrings: {all: familyArray[0], ranked: familyArray[1], unranked: familyArray[2]},
+                      MIBL: {minMax: mm, incDec: ir, belAbo: ba, lowHig: lh},
+                      isMM: {isMin, isMax}};
 } /* }}} */
 
-function rankedUnrankedString(argNameString) { /* {{{ */
-    if (argNameString.includes("ranked") && !argNameString.includes("unranked")) {
-        return "ranked";
-    } else if (argNameString.includes("unranked")) {
-        return "unranked";
+function beforeRankedUnrankedGamesSpecial(before, extra, argNameString, special) { /* {{{ */
+    const isExtra = (extra !== "");
+    if (argNameString.includes("unranked")) {
+        return `${before}${extra}unranked games`; //ex: "for blitz unranked games"
+    } else if (argNameString.includes("ranked")) {
+        return `${before}${extra}ranked games`;   //ex: "for ranked games"
+    } else if (isExtra) {
+        return `${before}${extra}games`           //ex: "for correspondence games"
+    } else if (special !== "") {
+        return `${before}${special}games`         //ex: "from all games"
     } else {
         return "";
     }
 } /* }}} */
 
-function checkAndGenerateArgNameString(familyArray, notificationRanked) { /* {{{ */
-    if (config[familyArray[1]] && notificationRanked) {
-        return familyArray[1];
-    } else if (config[familyArray[2]] && !notificationRanked) {
-        return familyArray[2];
+function allowedBannedSettingExtraRankedUnranked(allowedBannedSetting,argNameString) { /* {{{ */
+    if (argNameString.includes("unranked")) {
+        return `${allowedBannedSetting}_unranked`;
+    } else if (argNameString.includes("ranked")) {
+        return `${allowedBannedSetting}_ranked`;
     } else {
-        return familyArray[0];
+        return allowedBannedSetting;
+    }
+} /* {{{ */
+
+function checkAndGenerateArgNameStringObject(familyObjectArgNameStrings, notificationRanked) { /* {{{ */
```

Comment:
The return type here is a string, not a stringObject.

I'm used to languages that have types build in, but since Js doesn't do that you should at least keep naming consistent with the type returned.

Another strategy is to name things not after their type but what they are. So for example use plural form to indicate an array instead of adding `array` suffix. But again that might be more suited for typed languages.

Reply:
fixed in https://github.com/online-go/gtp2ogs/pull/209/commits/e9e6b222a8a595cea0ca3c61ff009343464bea9a
thanks ! :1st_place_medal: